internal/services/storage/migration/account.go

```diff
@@ -204,3 +211,810 @@ func accountSchemaForV0AndV1() map[string]*pluginsdk.Schema {
 		},
 	}
 }
+
+func accountSchemaForV2() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validate.StorageAccountName,
+		},
+
+		"resource_group_name": azure.SchemaResourceGroupName(),
+
+		"location": azure.SchemaLocation(),
+
+		"account_kind": {
+			Type:     pluginsdk.TypeString,
+			Optional: true,
+			ValidateFunc: validation.StringInSlice([]string{
+				string(storage.Storage),
+				string(storage.BlobStorage),
+				string(storage.BlockBlobStorage),
+				string(storage.FileStorage),
+				string(storage.StorageV2),
+			}, true),
+			Default: string(storage.StorageV2),
+		},
+
+		"account_tier": {
+			Type:     pluginsdk.TypeString,
+			Required: true,
+			ForceNew: true,
+			ValidateFunc: validation.StringInSlice([]string{
+				"Standard",
+				"Premium",
+			}, true),
+			DiffSuppressFunc: suppress.CaseDifference,
+		},
+
+		"account_replication_type": {
+			Type:     pluginsdk.TypeString,
+			Required: true,
+			ValidateFunc: validation.StringInSlice([]string{
+				"LRS",
+				"ZRS",
+				"GRS",
+				"RAGRS",
+				"GZRS",
+				"RAGZRS",
+			}, true),
+			DiffSuppressFunc: suppress.CaseDifference,
+		},
+
+		// Only valid for BlobStorage & StorageV2 accounts, defaults to "Hot" in create function
+		"access_tier": {
+			Type:     pluginsdk.TypeString,
+			Optional: true,
+			Computed: true,
+			ValidateFunc: validation.StringInSlice([]string{
+				string(storage.Cool),
+				string(storage.Hot),
+			}, true),
```

Comment:
fwiw you don't need these for a state migration - only the Type and Required/Optional/Computed iirc

```suggestion
```